### PR TITLE
Add RL scripts

### PR DESCRIPTION
**Pull request by andrewtanJS**

### CODE REVIEW

File: RL/cleanrl_connect_four.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        help="seed of the experiment")
+    parser.add_argument("--torch-deterministic", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
+        help="if toggled, `torch.backends.cudnn.deterministic=False`")
+    parser.add_argument("--cuda", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
+        help="if toggled, cuda will be enabled by default")
+    parser.add_argument("--save-model", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
+        help="whether to save model into the `runs/{run_name}` folder")
+
+    # Algorithm specific arguments
+    parser.add_argument("--env-id", type=str, default="BreakoutNoFrameskip-v4",
```
Comment (elliottower) [inline RL/cleanrl_connect_four.py:32]: Breakout code

File: RL/cleanrl_connect_four.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    action_mask = obs_dict["action_mask"]
+    print("Starting training:")
+    for global_step in range(args.total_timesteps):
+        # ALGO LOGIC: put action logic here
+        epsilon = linear_schedule(args.start_e, args.end_e, args.exploration_fraction * args.total_timesteps, global_step)
+        for agent in env.agent_iter():
+            if random.random() < epsilon:
+                actions = 0
+                while action_mask.any() and action_mask[actions] == 0:
+                    actions = np.array(action_space.sample())
```
Comment (elliottower) [inline RL/cleanrl_connect_four.py:142]: Just do sample(mask)

Comment (andrewtanJS) [reply RL/cleanrl_connect_four.py]: updated

File: RL/cleanrl_connect_four.py [hunk 3]
```diff
@@ -133,14 +133,13 @@ def linear_schedule(start_e: float, end_e: float, duration: int, t: int):
...
+            print("global step", global_step)
         # ALGO LOGIC: put action logic here
         epsilon = linear_schedule(args.start_e, args.end_e, args.exploration_fraction * args.total_timesteps, global_step)
         for agent in env.agent_iter():
             if random.random() < epsilon:
-                actions = 0
-                while action_mask.any() and action_mask[actions] == 0:
-                    actions = np.array(action_space.sample())
-                actions = np.array(actions)
+                actions = np.array(env.action_space(agent).sample(action_mask))
```
[the remark this answers is not in the document]
Comment (elliottower) [reply RL/cleanrl_connect_four.py:142]: The whole idea of sampling is it gives a valid action which can be used as is in the same format

**Comment by elliottower** [comment]
LGTM

**Comment by elliottower** [comment]
Merge whenever this is finished, don't want to prematurely merge if you're not done fully

**Pull request merged**